### PR TITLE
fix(calcite-label): label for clicks work properly inside shadowRoots

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** #1957

### CODE REVIEW

**Comment by eriklharper** [comment]
Keeping this in draft while I whip up a test for this.

**Comment by eriklharper** [comment]
For some reason I can't get this test to pass, so I'm pasting it here for reference just to keep a copy of the code.

```
it("clicking sibling label focuses input when both are inside a shadowRoot", async () => {
  const page = await newE2EPage();
  await page.evaluate(() => {
    class ShadowComponent extends HTMLElement {
      constructor() {
        super();
        const shadow = this.attachShadow({ mode: "open" });
        shadow.innerHTML = `
          <calcite-label for="input">Label</calcite-label>
          <calcite-input id="input"></calcite-input>
        `;
      }
    }
    customElements.define("shadow-component", ShadowComponent);
    const shadowComponent = document.createElement("shadow-component");
    document.body.appendChild(shadowComponent);
  });
  const inputFocusEvent = await page.spyOnEvent("calciteInputFocus");
  const calciteLabel = await page.find("shadow-component >>> calcite-label");
  expect(inputFocusEvent).toHaveReceivedEventTimes(0);
  await calciteLabel.click();
  await page.waitForChanges();
  expect(inputFocusEvent).toHaveReceivedEventTimes(1);
});
```

**Comment by driskull** [comment]
I'm not sure, but can you try running the click inside evaluate?

**Comment by jcfranco** [comment]
Just checked this out. You need to initialize the page w/ a dummy calcite-component to bootstrap the components. Otherwise, they'll never upgrade. Similar to this: https://github.com/Esri/calcite-components/blob/55b6affe737d9c40fe907c16554eccf326604edc/src/components/calcite-date-picker-month-header/calcite-date-picker-month-header.e2e.ts#L35-L38

**Comment by eriklharper** [comment]
I got the test working using the whole `window.eventDetail` trick that other tests do.  Also moving the click into the page.evaluate worked, but I think it would also work outside too, it was just the issue of listening to that event that was the biggest hurdle.  Should be good now!

File: src/demos/calcite-label.html [hunk 1]
```diff
@@ -17,6 +17,20 @@
 <script src="./_assets/head.js"></script>
   <script src="_assets/validateInput.js"></script>
   <script type="module" src="./_assets/demo-form.js"></script>
+  <script>
+    class ShadowComponent extends HTMLElement {
```
Comment (jcfranco) [inline src/demos/calcite-label.html:21]: I don't think we need this since it's covered by the unit test.

**Review by jcfranco — APPROVED** [review]
👍🏼

**Pull request merged**